ntbs-service/Services/LegacySearchBuilder.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Dynamic;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+using ntbs_service.Models;
+using ntbs_service.Models.Entities;
+
+namespace ntbs_service.Services
+{
+    public interface ILegacySearchBuilder : ISearchBuilder
+    {
+        (string, dynamic) GetResult();
+    }
+
+    public class LegacySearchBuilder : ILegacySearchBuilder
+    {
+        string sqlQuery;
+        dynamic parameters;
+        
+        public LegacySearchBuilder()
+        {
+            this.sqlQuery = "";
+            this.parameters = new ExpandoObject();
+        }
+
+        public ISearchBuilder FilterById(string id)
+        {
+            if (!string.IsNullOrEmpty(id))
+            {
+                sqlQuery += "WHERE n.OldNotificationId = @id OR n.GroupId = @id AND n.Source = 'LTBR' OR dmg.NhsNumber = @id";

```

Comment:
Where possible I'm matching on fields on the notifications view, I could move all these matches to match on the joined demographics view, this also then leads to a question of where should the indexes live (e.g. all matching on demographics here and indexes live on demographics view or split across multiple views). I'm not quite familiar enough with indexes to have a preference here

Reply:
My first instinct is that as long as we apply the index to the view that we are matching on, we are covering all important bases.
It might be that having a single view to filter and index on would be more performant, but I'm thoer-guessing here, so perhaps its worth following this approach first and watching for how the performance behaves!